tart2ms/tart2ms.py

```diff
@@ -756,41 +948,53 @@ def __twelveball(direction):
             'OBSERVATION_ID': (("row",), da.from_array(np.zeros(row, dtype=int), chunks=chunks['row'])),
             'STATE_ID': (("row",), da.from_array(np.zeros(row, dtype=int), chunks=chunks['row'])),
         }
-        if fill_model and sources:
+        if fill_model:
             main_table["MODEL_DATA"] = (dims, model_data)
         dataset = Dataset(main_table)
         ms_datasets.append(dataset)
 
     ms_writes = xds_to_table(ms_datasets, ms_table_name, columns="ALL")
+    # auxilary table futures creation
     spw_writes = xds_to_table(spw_datasets, spw_table_name, columns="ALL")
     ddid_writes = xds_to_table(ddid_datasets, ddid_table_name, columns="ALL")
+    for tt in [ant_table, feed_table, field_table, pol_table, obs_table, src_table]:
+        tt.write()
 
-    dask.compute(ms_writes)
-
-    ant_table.write()
-    feed_table.write()
-    field_table.write()
-    pol_table.write()
-    obs_table.write()
-    src_table.write()
-
-    dask.compute(spw_writes)
-    dask.compute(ddid_writes)
+    # execute graph with futures
+    LOGGER.info("Synthesizing MS...")
+    with ProgressBar():
+        dask.compute([ms_writes + spw_writes + ddid_writes] + MSTable.get_futures())
 
+    LOGGER.info("Performing finalization of UVW coordinates if needed")
     if uvw_generator == 'telescope_snapshot':
-        pass
+        pass # user has been warned about their choice -- this cannot be used when rephasing
```

Comment:
This generates an error above so no need to add an exception here as well for the phasing case